ui/panels-plugin/src/plugins/time-series-chart/TimeSeriesChartPanel.tsx

```diff
@@ -48,6 +48,8 @@ export function TimeSeriesChartPanel(props: TimeSeriesChartProps) {
 
   const suggestedStepMs = useSuggestedStepMs(contentDimensions?.width);
   const queryResults = useTimeSeriesQueries(queries, { suggestedStepMs });
+  const loading = queryResults.some((result) => result.isLoading);
+  const hasData = queryResults.some((result) => result.data && Array.from(result.data.series).length > 0);
```

Comment:
Changes:
- `useMemo` below will always attempt to crunch the data for the chart (i.e. `graphData`)
- `loading` and `hasData` will be used to decide when to show the loading state, and when to show an error state (see below)
- Errors are being caught by ErrorBoundary, which is a different discussion